scripts/azdo/update_sstate_cache.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+usage() {
+	local exit_code=${1:-2}
+	test $exit_code -eq 0 || exec 1>&2
+
+	cat <<EOF
+
+Syntax:
+$(basename $BASH_SOURCE) push local_sstate_cache_dir argo_sstate_cache_dir
+   Push the local sstate-cache directory contents to the remote argo
+   server directory.
+
+$(basename $BASH_SOURCE) fetch local_sstate_cache_dir argo_sstate_cache_dir
+   Fetch the latest sstate cache from the remote argo cache directory
+   and install to the local sstate-cache directory.
+
+$(basename $BASH_SOURCE) -h
+   Print this help message and exit
+
+Arguments:
+local_sstate_cache_dir      Path to the local sstate cache directory
+argo_sstate_cache_dir       Path to the remote argo sstate cache directory
+
+EOF
+	exit $exit_code
+}
+
+push_cache() {
+	if [ ! -d $local_sstate_cache_dir -o -z "$(ls -A $local_sstate_cache_dir 2>/dev/null)" ]; then
+		echo $local_sstate_cache_dir is missing/empty. Please provide a valid local_sstate_cache_dir name with sstate-cache contents.
+		usage
+	fi
+
+	mkdir -p $argo_sstate_cache_dir
+	pushd $argo_sstate_cache_dir >/dev/null
+
+	# Copy the contents of the cache to a temp .latest directory
+	# before removing the .latest suffix from the name.
+	temp_argo_dir=$(date +"%s").latest
+	mkdir $temp_argo_dir
+	echo Pushing sstate cache to argo at $temp_argo_dir
+	cp -rL $local_sstate_cache_dir/* $temp_argo_dir/
+	mv $temp_argo_dir ${temp_argo_dir%.latest}
+
+	# Cap the number of cache directories on the argo server at
+	# max_dir_count. Delete older cache directories exceeding the
+	# cap.
```

Comment:
There shouldn't be any .latest directories at this point. A .latest directory if created before would have already been renamed above.
I'm relying on `find` to count the number of directories before proceeding to delete. I could have used `ls`, but I didn't think of it then and don't think it will change the result now. I use `ls` later on to sort (by file modification time, not the name of the directory) and pick the directories to remove.

Reply:
>There shouldn't be any .latest directories at this point.

What if two pushes are being run concurrently?

>I use ls later on to sort (by file modification time, not the name of the directory) and pick the directories to remove.

Hmm.. I'm not sure we can rely on file modification time across the CIFS boundary. Since they're already named using UNIX-time, why not rely on that?

Reply:
> What if two pushes are being run concurrently?
Only the CI jobs will attempt pushes, and they are scheduled to run sequentially.

>Hmm.. I'm not sure we can rely on file modification time across the CIFS boundary. Since they're already named using UNIX-time, why not rely on that?
Sure.

Reply:
>Only the CI jobs will attempt pushes, and they are scheduled to run sequentially.

I agree that *shouldn't* happen, but I'm uncomfortable relying on that assumption. I'll defer to you as to whether or not the enhancement is worth it.